addons/osio-addon/src/test/java/io/fabric8/launcher/osio/preparers/InsertMavenDependenciesPreparerTest.java

```diff
@@ -0,0 +1,124 @@
+import io.fabric8.launcher.booster.catalog.rhoar.Mission;
+import io.fabric8.launcher.booster.catalog.rhoar.RhoarBooster;
+import io.fabric8.launcher.booster.catalog.rhoar.RhoarBoosterCatalogService;
+import io.fabric8.launcher.booster.catalog.rhoar.Runtime;
+import io.fabric8.launcher.booster.catalog.rhoar.Version;
+import io.fabric8.launcher.osio.projectiles.context.OsioProjectileContext;
+
+import java.nio.file.Path;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.maven.model.Model;
+import org.arquillian.smart.testing.rules.git.server.GitServer;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class InsertMavenDependenciesPreparerTest {
+    @ClassRule
+    public static GitServer gitServer = GitServer.bundlesFromDirectory("repos/boosters")
+            .usingPort(8765)
+            .create();
+
+    private static RhoarBoosterCatalogService boosterCatalogService;
+
+    @Rule
+    public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+
+    @BeforeClass
+    public static void setUp() throws ExecutionException, InterruptedException {
+        boosterCatalogService = buildDefaultCatalogService();
+        boosterCatalogService.index().get();
+    }
+
+    private static RhoarBoosterCatalogService buildDefaultCatalogService() {
+        if (boosterCatalogService == null) {
+            RhoarBoosterCatalogService.Builder builder = new RhoarBoosterCatalogService.Builder();
+            builder.catalogRepository(LauncherConfiguration.boosterCatalogRepositoryURI());
+            builder.catalogRef(LauncherConfiguration.boosterCatalogRepositoryRef());
+            builder.transformer(new TestRepoUrlFixer("http://localhost:8765"));
+            boosterCatalogService = builder.build();
+        }
+        return boosterCatalogService;
+    }
+
+    @Test
+    public void shouldAddDependencies() throws IOException, InterruptedException, URISyntaxException {
+        // given
+        final Optional<RhoarBooster> booster = boosterCatalogService.getBooster(
+                BoosterPredicates.withMission(new Mission("rest-http"))
+                        .and(BoosterPredicates.withRuntime(new Runtime("vert.x"))
+                                .and(BoosterPredicates.withVersion(new Version("redhat")))));
+
+
+        final RhoarBooster rhoarBooster = booster.get();
+        final Path rootPath = temporaryFolder.getRoot().toPath();
+        boosterCatalogService.copy(rhoarBooster, rootPath);
+
+        final InsertMavenDependenciesPreparer insertMavenDependenciesPreparer = new InsertMavenDependenciesPreparer();
```

Comment:
Although that is fine, it would be much easier if you just created a directory with a known pom.xml and passed null to the booster param (don't forget to mark it as @Nullable)

Reply:
Actually I take that back. Do you mind simplifying the test a bit? That looks unnecessarily complex now that I see it again